### PR TITLE
sql/parser: consistently handle binary operations with NULL

### PR DESCRIPTION
**Pull request by tamird**

### CODE REVIEW

**Comment by petermattis** [comment]
Add an eval test case that `NULL + 1` works.

LGTM

**Comment by tamird** [comment]
done.

File: sql/parser/type_check_test.go [hunk 1]
```diff
@@ -26,6 +26,12 @@ import (
 
 func TestTypeCheckExpr(t *testing.T) {
 	testData := []string{
+		`NULL + 1`,
+		`NULL + 1.1`,
+		`NULL + '2006-09-23'::date`,
+		`NULL + '1h'::interval`,
+		`NULL + 'hello'`,
+		`NULL + 'hello'::bytes`,
```
Comment (petermattis) [inline sql/parser/type_check_test.go]: Perhaps one of these should be `<constant> + NULL` (i.e. `NULL` on the right hand side).

Comment (tamird) [inline sql/parser/type_check_test.go]: I wrote the eval test that way.

Comment (petermattis) [inline sql/parser/type_check_test.go]: Ah, didn't notice that. Carry on.

**Pull request merged**